nep-12.mediawiki

```diff
@@ -0,0 +1,2098 @@
+<pre>
+  NEP: 12
+  Title: Unified dApp API for wallet providers
+  Author: Nick Fujita <nickfujita@gmail.com>, Matus Zamborsky <zamborsky@gmail.com>
+  Type: Standard
+  Status: Accepted
+  Created: 2018-08-28
+</pre>
+
+==Abstract==
+
+This NEP describes a common API interface for dApps to communicate with external wallet providers. The use of a trusted 3rd party wallet providers will help users feel more secure when using dApps, and the unified interface will help dApp creators to have a more uniform developer experience when making their dApps compatible with various providers.
+
+
+==Motivation==
+
+===End Users===
+
+As dApps come into the ecosystem, there will be more concerns about the safety of user assets. If dApps all required users to input their private keys in order to use them, it just takes one malicious dApp to steal all their funds. By using a trusted wallet provider which interfaces with the various dApps in the ecosystem on their behalf, users can reduce the exposure of their private keys. This will even allow users to transact with their hardware wallets via the wallet provider,  never having to reveal their private keys even to the wallet itself.
```

Comment:
well described!

There is an extra space after the comma: "(...) ,  never having (...)"